### DISCUSSION TITLE
Kalibreurstag 2024 / asymmetrical profiles

### DISCUSSION THREAD

**Discussion by sag-kalibreur** [pyroll-project/pyroll-core#253] [Show and tell]
### Hello folks, the 6th Kalibreurstag was a successful and interesting event.

The Jupyter notebooks presented were all very interesting. Especially the notebooks **numerical-assessment-rib-filling** and **axis_symmetric_rolling** with the merging of different contours with asymmetric profiles. I have already experimented with the latter and the results are really good.

**Good news:** They all work if you install the missing plugins and place the required files correctly in the directories :-)

1. pillar_model_intro.ipynb	
2. Recrystallisation_Brass_Wire.ipynb
3. H_Profiles_Neumann.ipynb	
4. axis_symmetric_rolling	
5. numerical-assessment-rib-filling.ipynbling.ipynb	
6. Planar_Geometry_Swedish_Oval.ipynb
7. Planar_Geometry_Z_Profile.ipynb

**Note:**
The notebook **Recrystallisation_Brass_Wire** only works in Pandas Export with the old version of Export, in which the stand orientations were not yet adjusted. Otherwise you get a key-error ‘orientation’.

[Uploading lendl-analysis-drop-like-profile.zip…]()

**Comment by QiangF** [comment]
Can you please upload those ipynb files? Thank you!

**Reply by RichardPfr** [reply]
You can find most of those notebooks in the [pyroll-examples](https://github.com/pyroll-project/pyroll-examples) Repository. The ones missing so far (axis_symmetric_rolling) we are soon going to upload there as well.

**Reply by RollingTechSupport** [reply]
Also all files can be found in the Zenodo archive https://doi.org/10.5281/zenodo.13450173.

**Comment by RichardPfr** [comment]
Thank you for your Feedback.

Regarding the Error in the `Recrystallisation_Brass_Wire` Notebook you mentioned:
When i install the newest Version of pyroll-export (2.2.0) i don't get any errors and the Notebook runs through. Can you reproduce the error or say for what version you get that error?

**Reply by sag-kalibreur** [reply]
Hello Richard,
I have taken another close look at the files. The error occurs because we had modified the export.py in the section `def to_pandas` so that the processing of the dataframe fits for our template notebook. With the original 2.2.0 the notebook runs through without errors.

**original V2.2.0:**
```
def to_pandas(sequence: Sequence[Unit]) -> pd.DataFrame:
"""
Exports a PyRolL PassSequence object to a pandas ``DataFrame``.
Uses a flattened version of the result of ``to_dict()``.
"""
df = pd.DataFrame([
_flatten_dict(_to_dict(u)) for u in sequence
])
df.sort_index(axis="columns", inplace=True)
return df.convert_dtypes()
```
**modified V2.2.0:**

```
def to_pandas(sequence: Sequence[Unit]) -> pd.DataFrame:
"""
Exports a PyRolL PassSequence object to a pandas ``DataFrame``.
Uses a flattened version of the result of ``to_dict()``.
"""
df = pd.DataFrame([
_flatten_dict(_to_dict(u)) for u in sequence
])

df.sort_index(axis="columns", inplace=True)

# Ausrichten verschiedener Konturen

df.loc[df['orientation']=='AntiY', ['in_profile_cross_section_y', 'out_profile_cross_section_y',
'in_profile_lendl_section_y', 'out_profile_lendl_section_y',
'usable_cross_section_x', 'usable_cross_section_y']] *= -1
# Ausrichten verschiedener Konturen

return df
```
We also modified the convert.py , but this has no effects on other processes...
